### DISCUSSION TITLE
Is it possible to add environment variables directly to compilation options when compiling Mxnet？

### DISCUSSION THREAD

**Discussion by xianyujie** [apache/incubator-mxnet#15348] [Q&A]
Hi, @pengzhao-intel , I have some questions to ask you.
```export OMP_NUM_THREADS=1```
```export MXNET_ENGINE_TYPE=NaiveEngine```
1. I want to add the above two lines of code in the process of compiling mxnet, are there any compilation options that can be set？
2. What compilation options are included in compiling mxnet？

many thanks...

[a comment by an automated account is not shown]

**Comment by TaoLv** [comment]
@xianyujie They are environmental variables which should be used at model runtime, not compilation stage.
For MXNet pypi packages, the corresponding compilation flags can be found in https://github.com/apache/incubator-mxnet/issues/8671

**Comment by vdantu** [comment]
@xianyujie : We would love to understand your use case a little better.  Why would you want to set the OMP_NUM_THREADS to 1? Are you using multiple instances of MXNet on the same host running in parallel? Is this an inference use case? You could always set these vars in the start-up script right? 

@mxnet-label-bot add [Question]

**Comment by xianyujie** [comment]
@vdantu I want to take part in the FRVT competition. They ask that mxnet runs in single-thread mode.
```
Implementations must run in single-threaded mode, because NIST will parallelize the test 
by dividing the workload across many cores and many machines. Implementations must ensure 
that there are no issues with their software being parallelized via the fork() function. 
Developers should take caution with checking threading when using third-party frameworks 
(e.g., TensorFlow, MXNet, etc.).
```
By setting the following steps, the mxnet could run in single-thread mode.
```
compile mxnet with OPENMP=0 
export OMP_NUM_THREADS=1
export MXNET_ENGINE_TYPE=NaiveEngine
```
How to make the value of MXNET_ENGINE_TYPE equal to NaiveEngine from the source code instead of using 'export'? I think it should be a switch_case or if_else statement to determine the value of type. Where is this code in the source code?

**Comment by vzhangmeng726** [comment]
@xianyujie   Do you find the code in the source code? thank you!!!

**Comment by TaoLv** [comment]
@xianyujie @vzhangmeng726 Try to force engine type to `NaiveEngine` here: https://github.com/apache/incubator-mxnet/blob/master/src/engine/engine.cc#L33
